dotCMS/src/main/java/com/dotcms/ai/domain/AIRequest.java

```diff
@@ -0,0 +1,219 @@
+        this.type = builder.type;
+        this.payload = builder.payload;
+        this.userId = builder.userId;
+    }
+
+    /**
+     * Creates a quick text AI request with the specified configuration, payload, and user ID.
+     *
+     * @param appConfig the application configuration
+     * @param payload the request payload
+     * @param userId the user ID
+     * @param <T> the type of the request payload
+     * @param <R> the type of the AIRequest
+     * @return a new AIRequest instance
+     */
+    public static <T extends Serializable, R extends AIRequest<T>> R quickText(final AppConfig appConfig,
+                                                                               final T payload,
+                                                                               final String userId) {
+        return quick(AIModelType.TEXT, appConfig, payload, userId);
+    }
+
+    /**
+     * Creates a quick image AI request with the specified configuration, payload, and user ID.
+     *
+     * @param appConfig the application configuration
+     * @param payload the request payload
+     * @param userId the user ID
+     * @param <T> the type of the request payload
+     * @param <R> the type of the AIRequest
+     * @return a new AIRequest instance
+     */
+    public static <T extends Serializable, R extends AIRequest<T>> R quickImage(final AppConfig appConfig,
+                                                                                final T payload,
+                                                                                final String userId) {
+        return quick(AIModelType.IMAGE, appConfig, payload, userId);
+    }
+
+    /**
+     * Creates a quick embeddings AI request with the specified configuration, payload, and user ID.
+     *
+     * @param appConfig the application configuration
+     * @param payload the request payload
+     * @param userId the user ID
+     * @param <T> the type of the request payload
+     * @param <R> the type of the AIRequest
+     * @return a new AIRequest instance
+     */
+    public static <T extends Serializable, R extends AIRequest<T>> R quickEmbeddings(final AppConfig appConfig,
+                                                                                     final T payload,
+                                                                                     final String userId) {
+        return quick(AIModelType.EMBEDDINGS, appConfig, payload, userId);
+    }
+
+    public static <T extends Serializable, B extends AIRequest.Builder<T, B>> Builder<T, B> builder() {
+        return new Builder<>();
+    }
+
+    /**
+     * Resolves the URL for the specified model type and application configuration.
+     *
+     * @param type the AI model type
+     * @param appConfig the application configuration
+     * @return the resolved URL
+     */
+    static String resolveUrl(final AIModelType type, final AppConfig appConfig) {
+        switch (type) {
```

Comment:
this could be a Map

resolveUrlMap.put (TEXT, appConfig.getApiUrl()); 
...

resolveUrlMap.get(type):

Reply:
Storing into a String variable instead.